RegExpSupport/src/org/intellij/lang/regexp/RegExpLanguageHost.java

```diff
@@ -14,7 +14,21 @@ public interface RegExpLanguageHost {
   EnumSet<RegExpGroup.Type> EMPTY_NAMED_GROUP_TYPES = EnumSet.noneOf(RegExpGroup.Type.class);
   String[][] EMPTY_COMPLETION_ITEMS_ARRAY = new String[0][];
 
+  /**
+   * @deprecated Use {@link #characterNeedsEscaping(char, boolean)} instead.
+   */
+  @Deprecated
```

Comment:
I presume default implementation should be added here. Otherwise I can't get rid of implementing deprecated method, which feels wrong. 

cc @BasLeijdekkers

Reply:
My two cents (and the reason I wrote it this way): adding the new method without a default implementation would amount to a breaking change, and anyone downstream who pulled the latest changes would immediately get a build breakage.

I suppose both methods could be given empty default implementations that just return false, which might serve both purposes.

I'm not fully aware on guidelines around making breaking changes, so don't have a specific recommendation. Just thought I'd add that context.

Reply:
Not sure I got your idea.
My point is that you should have add default implementation for deprecated method with AbstractMethodException inside.